### ISSUE TITLE
updateOrCreate() fails to respect manaully set primary key

### ISSUE THREAD

**Issue by gregrobson** [laravel/framework#24175]
- Laravel Version: 5.6.12 – 5.6.21 (possibly existed earlier than this)
- PHP Version: 7.1 (although I imagine any 5.6 compatible versions will be affected)
- Database Driver & Version: PostgreSQL 10

### Description:
If you use `updateOrCreate()` on a model that uses a different primary key than is the model default the insert statement tries to return the primary key by using Laravel's assumed primary key name.

### Steps To Reproduce:

With a simple model

```php
class SatisfactionRating extends Model
{
    protected $table = 'freshdesk_satisfaction_rating'; // Custom table name
    protected $primaryKey = 'rating_id'; // Custom primary key

    protected $casts = ['rating' => 'array'];
}
```

Attempt an `updateOrCreate()`...

```php
SatisfactionRating::updateOrCreate(
    ['satisfaction_rating_id' => 42],
    ['rating' => []]
);
```

**Resulting error (abbreviated)**
(SQL: insert into "freshdesk_satisfaction_rating" ("rating_id", "rating") values (...) **returning "satisfaction_rating_id"**) {"exception":"[object] (Illuminate\\Database\\QueryException(code: 42703): SQLSTATE[42703]: Undefined column: 7 ERROR:  column "satisfaction_rating_id" does not exist
15639 LINE 1: ..., ) values ($1, $2) returning "satisfact...

Note section in bold. The query is assuming the returning column is `satisfaction_rating_id` not `rating_id` as specified in the model.

**Comment by Jamesking56** [comment]
Possibly an issue with this line:

https://github.com/laravel/framework/blob/5.6/src/Illuminate/Database/Query/Grammars/PostgresGrammar.php#L109

**Comment by tillkruss** [comment]
Can you submit a PR to fix this?

**Comment by kevdotbadger** [comment]
So, does `satisfaction_rating_id`, `rating_id` and `rating` exist on your model?

`updateOrCreate()` will fail because by the looks of things, `satisfaction_rating_id` doesn't exist?

Post your full(ish) error and migration.

**Comment by gregrobson** [comment]
@Jamesking56 I've traced that code back through the stack and I'm 90% certain I know what's happening.
As PostgreSQL favours `model_name_id` as primary key instead of `id` I call to `$this->setKeyName()` in the constructor of a class that extends the default model with a couple of custom behaviours that differ from Laravel's defaults. I think the way I'm calling `setKeyName()` is overwriting the property I set on the class.

Give me 24 hours to investigate, it will be the first thing on my list when I get in tomorrow,  I'll close this issue out if that's the case.

_Of all the people to comment first it would have to be you @Jamesking56!_ [FYI: previous colleague]👋
